src/metaswitch/clearwater/config_manager/config_type_class_plugin.py

```diff
@@ -0,0 +1,130 @@
+    """
+
+    def __init__(self, configpath):
+        """
+        self.scripts - A list of lists, this is needed because the lists are
+        passed into subprocess.call to run the validation script. They are in
+        a list as multiple validation scripts can be available for a
+        configuration type.
+        The config types can be broken down into categories and depending on
+        which category the validation scripts are a certain style.
+        self.configfile - The path to the config file to be validated.
+        """
+        self.configfile = configpath
+
+        if self.filetype == 'json':
+            self.scripts = self.get_json_validation()
+
+        elif self.filetype == 'xml':
+            self.scripts = self.get_xml_validation()
+
+        elif self.filetype == 'shared_config':
+            self.scripts = self.get_sharedconfig_validation()
+
+        else:
+            # Currently weatherwax has no validation scripts
+            self.scripts = {}
+
+    def __str__(self):
+        """If the class is asked to be printed it's name is printed."""
+        return self.name
+
+    def get_json_validation(self):
+        """Returns the scripts to be used for json file validation."""
+        scripts = {}
+        scripts[self.schema] = ['python', JSON_GENERIC_VALIDATE, self.schema,
+                                self.configfile]
+        # to add more validation scripts add to the dict of scripts
+        return scripts
+
+    def get_xml_validation(self):
+        """Returns the scripts to be used for xml file validation."""
+        scripts = {}
+        scripts[self.schema] = ['xmllint', '--format', '--pretty', '1',
+                                '--debug', '--schema',
+                                '{}'.format(self.schema),
+                                '{}'.format(self.configfile)]
+        # to add more validation scripts add to the dict of scripts
+        return scripts
+
+    def get_sharedconfig_validation(self):
+        """Returns the scripts to be used for shared config file validation."""
+        script_dir = glob.glob(os.path.join(VALIDATION_SCRIPTS_FOLDER, '*'))
+        # We can only execute scripts that have execute permissions.:q
+        scripts = {script: [script, self.configfile]
+                   for script in script_dir
+                   if os.access(script, os.X_OK)}
+        return scripts
+
+    def validate(self):
+        """
+        This method inherits the scripts to run validation against and then
+        runs them all. To run them the scripts must be able to be passed into
+        subprocess.check_output().
+        Then returning a list of those that have failed.
+        """
+
+        failed_scripts = []
+        error_lines = []
+        for script in self.scripts:
+            try:
+                log.debug("Running validation script %s", script)
+                subprocess.check_output(self.scripts[script], stderr=subprocess.STDOUT)
```

Comment:
When you are looping over a dictionary and have to do something like this, it's often cleaner to re-write the loop like this:
```
for name, script in self.scripts.items():
    function_that_uses_script(script)
    function_that_uses_name(name)
```
The `items()` method turns the dictionary into a list of tuples, making the loop look a bit nicer.

It's not a big deal though - at this point it's not worth fixing.